example/randomCells2D/src/randomCells2D.cpp

```diff
@@ -0,0 +1,187 @@
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <alpaka/rand/RandPhilox.hpp>  //XXX: I don't like this.
+
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+
+    unsigned constexpr NUM_CALCULATIONS = 256;
+    unsigned constexpr NUM_X = 100;
+    unsigned constexpr NUM_Y = 100;
+
+
+    struct InitRandomKernel
+    {
+        template<typename TAcc, typename TExtent, typename TRandEngine>
+        ALPAKA_FN_ACC auto operator()(TAcc const& acc, TExtent const extent, TRandEngine* const states) const -> void
+        {
+            auto idx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+            auto linearIdx = alpaka::mapIdx<1u>(idx, extent)[0];
+            TRandEngine engine(42, static_cast<std::uint32_t>(linearIdx));
+            states[linearIdx] = engine;
+        }
+    };
+
+    struct RunTimestepKernelSingle
+    {
+        template<typename TAcc, typename TExtent>
+        ALPAKA_FN_ACC auto operator()(
+            TAcc const& acc,
+            TExtent const extent,
+            alpaka::rand::Philox4x32x10<TAcc>* const states,
+            float* const cellsA) const -> void
+        {
+            auto idx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+            auto linearIdx = alpaka::mapIdx<1u>(idx, extent)[0];
+
+            // Setup generator and distribution.
+            alpaka::rand::Philox4x32x10<TAcc> engine(states[linearIdx]);
+            alpaka::rand::UniformReal<float> dist;
+
+            float sumA = 0;
+            for(unsigned numCalculations = 0; numCalculations < NUM_CALCULATIONS; ++numCalculations)
+            {
+                sumA += dist(engine);
+            }
+            cellsA[linearIdx] = sumA / NUM_CALCULATIONS;
+            states[linearIdx] = engine;
+        }
+    };
+
+    struct RunTimestepKernelVector
+    {
+        template<typename TAcc, typename TExtent>
+        ALPAKA_FN_ACC auto operator()(
+            TAcc const& acc,
+            TExtent const extent,
+            alpaka::rand::Philox4x32x10Vector<TAcc>* const states,
+            float* const cellsB) const -> void
+        {
+            auto idx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+            auto linearIdx = alpaka::mapIdx<1u>(idx, extent)[0];
+
+            // Setup generator and distribution.
+            alpaka::rand::Philox4x32x10Vector<TAcc> engine(states[linearIdx]);
+            unsigned constexpr resultVectorSize = std::tuple_size_v<decltype(engine())>;
```

Comment:
Do you really need to go through `std::tuple_size_v`? Does `engine()` return a `std::array`? Because then you could use:
`constexpr auto resultVectorSize = decltype(engine())::size();`

Consider `engine` also having some kind of `size()` member, if that information is so critical.